
diff --git a/isl/isl_map_simplify.c b/isl/isl_map_simplify.c
--- a/isl/isl_map_simplify.c	2018-11-20 14:10:56.345699488 -0800
+++ b/isl/isl_map_simplify.c	2018-11-20 14:11:16.522062268 -0800
@@ -733,7 +733,7 @@
 
 	elim_for = isl_calloc_array(ctx, int, bmap->n_div);
 	size = round_up(4 * bmap->n_div / 3 - 1);
-	bits = ffs(size) - 1;
+	bits = __builtin_ffs(size) - 1;
 	index = isl_calloc_array(ctx, int, size);
 	if (!index)
 		return bmap;
@@ -1124,7 +1124,7 @@
 		return bmap;
 
 	size = round_up(4 * (bmap->n_ineq+1) / 3 - 1);
-	bits = ffs(size) - 1;
+	bits = __builtin_ffs(size) - 1;
 	ctx = isl_basic_map_get_ctx(bmap);
 	index = isl_calloc_array(ctx, isl_int **, size);
 	if (!index)
@@ -1729,7 +1729,7 @@
 		return NULL;
 
 	size = round_up(4 * (context->n_ineq+1) / 3 - 1);
-	bits = ffs(size) - 1;
+	bits = __builtin_ffs(size) - 1;
 	ctx = isl_basic_set_get_ctx(bset);
 	index = isl_calloc_array(ctx, isl_int **, size);
 	if (!index)

diff --git a/isl/isl_hash.c b/isl/isl_hash.c
--- a/isl/isl_hash.c	2018-11-20 14:11:22.738174000 -0800
+++ b/isl/isl_hash.c	2018-11-20 14:11:33.866373983 -0800
@@ -50,7 +50,7 @@
 
 	if (min_size < 2)
 		min_size = 2;
-	table->bits = ffs(round_up(4 * (min_size + 1) / 3 - 1)) - 1;
+	table->bits = __builtin_ffs(round_up(4 * (min_size + 1) / 3 - 1)) - 1;
 	table->n = 0;
 
 	size = 1 << table->bits;
