diff --git a/libgcc/unwind.inc b/libgcc/unwind.inc
index 87e2ef0..3683936 100644
--- a/libgcc/unwind.inc
+++ b/libgcc/unwind.inc
@@ -78,10 +78,13 @@ _Unwind_RaiseException_Phase2(struct _Unwind_Exception *exc,
 
 /* Raise an exception, passing along the given exception object.  */
 
+extern void blah1 (int,int,const char*);
+
 _Unwind_Reason_Code LIBGCC2_UNWIND_ATTRIBUTE
 _Unwind_RaiseException(struct _Unwind_Exception *exc)
 {
   struct _Unwind_Context this_context, cur_context;
+  blah1(1,sizeof(this_context),NULL);
   _Unwind_Reason_Code code;
 
   /* Set up this_context to describe the current stack frame.  */
