diff --git a/newlib/libc/misc/lock.c b/newlib/libc/misc/lock.c
index c03c36a1e..8a2a0ba13 100644
--- a/newlib/libc/misc/lock.c
+++ b/newlib/libc/misc/lock.c
@@ -1,3 +1,4 @@
+#if 0
 /*
 FUNCTION
 <<__retarget_lock_init>>, <<__retarget_lock_init_recursive>>, <<__retarget_lock_close>>, <<__retarget_lock_close_recursive>>, <<__retarget_lock_acquire>>, <<__retarget_lock_acquire_recursive>>, <<__retarget_lock_try_acquire>>, <<__retarget_lock_try_acquire_recursive>>, <<__retarget_lock_release>>, <<__retarget_lock_release_recursive>>---locking routines
@@ -151,3 +152,4 @@ __retarget_lock_release_recursive (_LOCK_T lock)
 }
 
 #endif /* !defined(__SINGLE_THREAD__) */
+#endif
